src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Main.cs

```diff
@@ -1261,7 +1261,7 @@ public class MyOptions
 
             // validate the location is inside the MyOptions class and not outside the compilation which is in the referenced assembly
             Assert.StartsWith("src-0.cs: (12,", diag.Location.GetLineSpan().ToString());
-        }, assemblyPath).Dispose();
+        }, assemblyPath, new RemoteInvokeOptions { TimeOut = 300 * 1000}).Dispose();
```

Comment:
nit: save the options to a static/local since it's used again below?

Also how did you land on this number?

Reply:
> nit: save the options to a static/local since it's used again below?

I think it is not worth it to do so. Also, we may decide to change the timeout on different tests later. I'll watch if we add more and consider this optimization at that time. 

> Also how did you land on this number?

The default timeout for remote executor is 1 minute. I made this 5 minute to have enough time especially under the stress environment which can delay letting the remote process to compete.